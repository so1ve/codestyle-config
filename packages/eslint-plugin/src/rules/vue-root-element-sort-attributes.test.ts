import type { InvalidTestCase, ValidTestCase } from "eslint-vitest-rule-tester";
import { unindent as $ } from "eslint-vitest-rule-tester";
import { expect } from "vitest";
import VueEslintParser from "vue-eslint-parser";

import { run } from "./_test";
import type { Options } from "./vue-root-element-sort-attributes";
import rule, { RULE_NAME } from "./vue-root-element-sort-attributes";

const valid: ValidTestCase<Options>[] = [
	$`
	  <script setup lang="ts"></script>
	`,
];
const invalid: InvalidTestCase<Options>[] = [
	'<script lang="ts" setup></script>',
	{
		code: $`
		  <script lang="ts" setup></script>
		  <template b a></template>
		`,
		options: [
			{
				script: ["lang", "setup"],
				template: ["a", "b"],
			},
		],
		output: (o) => expect(o).toMatchSnapshot(),
	},
];

run({
	name: RULE_NAME,
	rule,
	parser: VueEslintParser,
	valid,
	invalid,
});
