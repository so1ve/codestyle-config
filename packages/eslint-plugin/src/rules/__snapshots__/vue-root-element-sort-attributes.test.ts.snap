// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vue-sort-root-element-attributes > invalid > Invalid #0: <script lang="ts" setup></script> 1`] = `"<script   setup lang="ts"></script>"`;

exports[`vue-sort-root-element-attributes > invalid > Invalid #1: <script lang="ts" setup></script>
<template b a></template> 1`] = `
"<script lang="ts" setup></script>
<template   a b></template>"
`;
